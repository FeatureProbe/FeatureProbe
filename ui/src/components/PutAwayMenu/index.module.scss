.putaway {
  height: 32px;
  line-height: 1;
  display: flex;
  align-items: center;
}

.iconfont {
  font-size: 18px !important;
  vertical-align: middle;
}

.menu {
  vertical-align: middle;
  margin-left: 8px;
}