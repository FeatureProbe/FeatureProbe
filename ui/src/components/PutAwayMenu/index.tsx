import { Popup } from 'semantic-ui-react';
import Icon from 'components/Icon';
import styles from './index.module.scss';

interface IProps {
  isPutAway: boolean;
  title: string;
  type: string;
}

const PutAwayMemu = (props: IProps) => {
  const { isPutAway, title, type } = props;

  return (
    <div className={styles.putaway}>
      {
        isPutAway ? (
          <>
            <Popup 
              inverted
              position='right center'
              offset={[0, 20]}
              content={title} 
              style={{opacity: 0.8}}
              trigger={
                <Icon customclass={styles['iconfont']} type={type} />
              } 
            />
          </>
        ) : (
          <>
            <Icon customclass={styles['iconfont']} type={type} />
            <span className={styles.menu}>
              {title}
            </span>
          </>
        )
      }
    </div>
	);
};

export default PutAwayMemu;