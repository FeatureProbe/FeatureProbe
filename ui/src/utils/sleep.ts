const sleep = (delay: number) => {
  return new Promise((res) => {
    setTimeout(() => {
      res('ok');
    }, delay);
  });
};

export default sleep;
