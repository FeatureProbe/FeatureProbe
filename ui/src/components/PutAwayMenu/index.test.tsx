import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import PutAwayMenu from '.';
import sleep from 'utils/sleep';

console.error = jest.fn();

it('PutAwayMenu snapshot', (done) => {
  (async () => {
    const { asFragment, rerender } = render(<PutAwayMenu isPutAway={true} title="title" type="type" />);
    expect(asFragment()).toMatchSnapshot();
    rerender(<PutAwayMenu isPutAway={false} title="title" type="type" />);
    expect(asFragment()).toMatchSnapshot();
    done();
  })();
});

test('hover PutAwayMenu', (done) => {
  (async () => {
    const { baseElement } = render(<PutAwayMenu isPutAway={true} title="title" type="type" />);
    const ele = screen.getAllByRole('generic').pop();
    expect(ele).not.toBeUndefined();
    
    if(ele) {
      await userEvent.hover(ele);
      await sleep(1000);
    }

    expect(baseElement).toMatchSnapshot();

    done();
  })();
});
