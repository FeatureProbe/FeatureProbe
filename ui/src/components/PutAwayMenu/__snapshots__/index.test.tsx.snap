// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PutAwayMenu snapshot 1`] = `
<DocumentFragment>
  <div
    class="putaway"
  >
    <span
      class="iconfont icon-type iconfont"
      customclass="iconfont"
      type="type"
    />
  </div>
</DocumentFragment>
`;

exports[`PutAwayMenu snapshot 2`] = `
<DocumentFragment>
  <div
    class="putaway"
  >
    <span
      class="iconfont icon-type iconfont"
      customclass="iconfont"
      type="type"
    />
    <span
      class="menu"
    >
      title
    </span>
  </div>
</DocumentFragment>
`;

exports[`hover PutAwayMenu 1`] = `
<body>
  <div>
    <div
      class="putaway"
    >
      <span
        class="iconfont icon-type iconfont"
        customclass="iconfont"
        type="type"
      />
    </div>
  </div>
  <div
    style="display: flex; position: absolute; left: 0px; top: 0px; transform: translate(20px, 0px);"
  >
    <div
      class="ui right center inverted popup transition visible"
      style="position: initial; opacity: 0.8;"
    >
      <div
        class="content"
      >
        title
      </div>
    </div>
  </div>
</body>
`;
